### PR TITLE
Fix #8: Initiate a maven invoker builder class and add corresponding dependencies.

### LINKED ISSUE #8: Property: compilation part1-Initiate the Maven builder class
After discussing, Emil and I decided to use the Maven invoker to finish the compilation.
Then we need the corresponding dependencies and a basic class.

### PR DESCRIPTION
**Pull request by Tian0602**
This class contains an instantiate and basic function calling. It might be used directly after the cloning part is finished.

### CODE REVIEW

File: DD2480-CI/src/main/java/mavenBuilder/MavenInvokerBuilder.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+     * Define a field for the calling program instance.
+     **/
+    private final Invoker invoker;
+
+    /**
+     * Instantiating the calling program in the class constructor
+     **/
+    public MavenInvokerBuilder(File localRepositoryDir) {
+        this.invoker = new DefaultInvoker();
+        this.invoker.setLocalRepositoryDirectory(localRepositoryDir);
```
Comment (Frequinzy) [inline DD2480-CI/src/main/java/mavenBuilder/MavenInvokerBuilder.java:27]: Does this work? Invoker is final so it can't be changed once initialized right?

**Review by Frequinzy — CHANGES_REQUESTED** [review]
Should include tests before merging

File: DD2480-CI/src/main/java/mavenBuilder/MavenInvokerBuilder.java (hunk 1 above)
Comment (Tian0602) [reply DD2480-CI/src/main/java/mavenBuilder/MavenInvokerBuilder.java:24]: Yeah, it works. I think "final" just means this invoker won't be another invoker. But it is okay to set attributes for this one. Btw, tests were added.

Comment (Frequinzy) [reply DD2480-CI/src/main/java/mavenBuilder/MavenInvokerBuilder.java:24]: That makes sense

**Review by filippanilsson — APPROVED** [review]

**Review by Frequinzy — APPROVED** [review]

**Pull request merged**